### PR TITLE
Remove meters from child registries using mapped Id

### PR DESCRIPTION
**Pull request by shakuzen**
Child registries may have `MeterFilter` configured that alter the Id of meters when registered. In this case, calling `remove` with the Id from the `CompositeMeterRegistry` will not match unless it also has the same `MeterFilter` configured. The behavior of the `remove` method has fluctuated as these cases come to light. This adds public API for removing a meter with its pre-mapped Id and clarifies the behavior in the JavaDoc of all `remove` methods. `CompositeMeterRegistry` can now use this new API to have the expected behavior.

Resolves gh-2354

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java [hunk 1]
```diff
@@ -614,15 +617,29 @@ public Meter remove(Meter meter) {
...
-    Meter remove(Meter.Id id, boolean applyFilters) {
+    public Meter remove(Meter.Id id, boolean applyFilters) {
         if (applyFilters) {
             return remove(getMappedId(id));
         }
         return remove(id);
     }
 
     /**
+     * Remove a {@link Meter} from this {@link MeterRegistry registry} based on its {@link Id} after any
```
Comment (shakuzen) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java:639]: Re-reading this, it's not as clear as I would like... let me try again.

File: micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java [hunk 2]
```diff
@@ -614,15 +617,29 @@ public Meter remove(Meter meter) {
...
+     *
+     * @param id the id of the meter to remove
+     * @param applyFilters whether to apply this registry's filters to the id before attempting to remove
+     * @return The removed meter, or null if the meter is not found
+     * @since 1.3.16
+     */
+    @Incubating(since = "1.3.16")
     @Nullable
-    Meter remove(Meter.Id id, boolean applyFilters) {
+    public Meter remove(Meter.Id id, boolean applyFilters) {
```
Comment (shakuzen) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java:631]: Since this will add public API, I was thinking about it more, and I'm not sure I like this signature, given passing `false` is the same behavior as not passing the parameter at all (existing methods). I'm wondering if it is better to make a different method name. Therein lies the naming bikeshed, though.
* `removeByPreFilterId(Meter.Id preFilterId)`
Something along the lines of this is what I'm leaning towards.

Comment (admoca60) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java:631]: Ok, it could be Ok, but in that case, don't forget to modify the "CompositeMeterRegistry.java" file to replace `r -> r.remove(m.getId(), true))` per `r -> r.removeByPreFilterId(m.getId()))`
